src/tasks/shop-admin/Rule/AssignEntitiesToRule.ts

```diff
@@ -0,0 +1,57 @@
+import {Locator, test as base} from '@playwright/test';
+import type {Task} from '../../../types/Task';
+import type {FixtureTypes} from '../../../types/FixtureTypes';
+
+interface Entity {
+    id: string;
+    name: string;
+    ruleType: string;
+}
+export const AssignEntitiesToRule = base.extend<{ AssignEntitiesToRule: Task }, FixtureTypes>({
+    AssignEntitiesToRule: async ({ AdminRuleDetail }, use ) => {
+        const task = (assignableEntities: Entity[]) => {
+            return async function AssignEntitiesToRule() {
+                async function entityAssignment(entityName: string, card: Locator) {
+                        const entityCard = await AdminRuleDetail.getEntityCard(card);
+                        await entityCard.addAssignmentButton.click();
+                        await AdminRuleDetail.page.locator('.sw-settings-rule-add-assignment-modal').getByPlaceholder('Search...').fill(entityName);
+                        await AdminRuleDetail.page.locator('.sw-data-grid__row').filter({hasText: entityName}).getByRole('checkbox').click();
+                        await AdminRuleDetail.page.locator('.sw-button--primary').getByText('Add').click();
+                }
+                for (const entity of assignableEntities) {
+                    const validRuleType = {
+                        shipping1: 'shippingMethodAvailabilityRule',
+                        tax1: 'taxProviderRule',
+                        payment1: 'paymentMethodAvailabilityRule',
+                        promotion1: 'promotionOrderRule',
+                        promotion2: 'promotionCustomerRule',
+                        promotion3: 'promotionCartRule',
+                    };
```

Comment:
Why is this loop necessary? Couldn't you just use the values directly in the case statements? And if it is necessary, I would probably use better understandable naming like shippingMethod, taxProvider, promotionOrderRule instead of shipping1, tax1, promotion1 etc.

Reply:
valid point